taf/updater/updater.py

```diff
@@ -195,8 +204,18 @@ def _update_target_repositories(repositories, repositories_json, repositories_co
     else:
       repository.fetch(True)
 
+    if allow_unauthenticated and old_head is not None:
```

Comment:
You can move first part of this if-statement under below one (`if old_head is not None:`).